perun/collect/complexity/run.py

```diff
@@ -39,174 +38,206 @@
+    except (OSError, subprocess.CalledProcessError) as exception:
+        print('Failed.\n')
+        return 1, str(exception), kwargs
 
 
 def after(**kwargs):
-    """ Handles the complexity collector post processing
-
-    Arguments:
-        kwargs(dict): dictionary containing the configuration settings for the complexity collector
+    """ Handles the complexity collector output and transforms it into resources
 
-    Returns:
-        tuple: int as a status code, nonzero values for errors
-               string as a status message, mainly for error states
-               dict of modified kwargs with bin value representing the executable
+    :param dict kwargs: dictionary containing the configuration settings for the collector
+    :returns: tuple (int as a status code, nonzero values for errors,
+                    string as a status message, mainly for error states,
+                    dict of modified kwargs with 'profile' containing the processed trace)
     """
-    # Get the trace log path
-    print('Starting the post-processing phase...')
-    pos = kwargs['cmd'].rfind('/')
-    path = kwargs['cmd'][:pos + 1] + kwargs['internal_data_filename']
-    address_map = symbols.extract_symbol_address_map(kwargs['cmd'])
 
+    print('Starting the post-processing phase... ', end='')
     resources, call_stack = [], []
-    profile_start, profile_end = 0, 0
-
-    with open(path, 'r') as profile:
-
-        is_first_line = True
-        for line in profile:
-            # Split the line into action, function name, timestamp and size
-            record = _ProfileRecord(*line.split())
-
-            # Process the record
-            if _process_file_record(record, call_stack, resources, address_map) != 0:
-                # Stack error
-                err_msg = 'Call stack error, record: ' + record.func + ', ' + record.action
-                if not call_stack:
-                    err_msg += ', stack top: empty'
-                else:
-                    err_msg += ', stack top: ' + call_stack[-1].func + ', ' + call_stack[-1].action
-                return 1, err_msg, kwargs
-
-            # Get the first and last record timestamps to determine the profiling time
-            profile_end = record.timestamp
-            if is_first_line:
-                is_first_line = False
-                profile_start = record.timestamp
-
-    # Update the profile dictionary
-    kwargs['profile'] = {
-        'global': {
-            'time': str((int(profile_end) - int(profile_start)) / _MICRO_TO_SECONDS) + 's',
-            'resources': resources
-        }
-    }
-    print('Done.\n')
-    return 0, _COLLECTOR_STATUS_MSG[0], dict(kwargs)
+    func_map = dict()
 
+    # Get the trace log path
+    try:
+        with open(kwargs['output'], 'r') as profile:
+
+            # Create demangled counterparts of the function names
+            demangler = shutil.which('c++filt')
+            if demangler is None:
```

Comment:
demangler is None -> not demangler?

Reply:
Also codacy does not like this, so would be good to call shlex.quote to guard against potential attack.